src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -1244,6 +1268,36 @@ caseSList
         {#caseSList = #(#[SLIST,"SLIST"],#caseSList);}
     ;
 
+switchRule
+    :   (       (switchLabeledExpr)=>     se:switchLabeledExpr
+        |       (switchLabeledBlk)=>      sb:switchLabeledBlk
+        |       (switchLabeledThrow)=>    st:switchLabeledThrow
+        )
+        {## = #(#[CASE_GROUP, "CASE_GROUP"], se, sb, st);}
```

Comment:
Should we re-use `CASE_GROUP`, as I've done here, or make a new token? Something like `SWITCH_RULE`?  Reusing CASE_GROUP will make check updates simpler, but would anything be gained by creating a new token?

Reply:
This is a good question. I see at least a problem with [FallThrough](https://checkstyle.org/config_coding.html#FallThrough) check. The Java 14 switch expression cases may not contain a `break`, so almost all of them will be reported by this check. And it seems that writing the utility method `isItCaseGroupYouAreLookingFor` will be difficult. By adding a new token instead of case group this problem goes away. Another one appears - in some cases,`case_group` will be skipped, but it will be easier to fix, usually by simply adding one `logical or`.

Reply:
Good callout @nmancus1, what you said makes sense to me @pbludov

Reply:
@romani What do you think?

Reply:
I have introduced the new token, SWITCH_RULE,  for the following reasons:

1. From[ JLS](https://docs.oracle.com/javase/specs/jls/se13/preview/switch-expressions.html#jep354-14.11.1): _When a selector expression matches a switch label for a switch labeled rule, the labeled expression or statement is executed and nothing else._

    This means that there is no reason to check for fall-through, as in the case of switch case groups.  In checks, we can simply conclude that we are in a switch rule, and know that there will be definite assignment and no fall through. 

2. The java grammar differentiates between the two, so we should also. 

    **_The body of both a switch statement and a switch expression (15.28) is known as a switch block. This block can consist of either:_**

    _Switch labeled rules, which use -> to introduce either a switch labeled expression, switch labeled block, or a switch labeled throw statement; or_

    _Switch labeled statement groups, which use : to introduce switch labeled block statements._